javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/config/ConfigPropertySource.java

```diff
@@ -16,19 +16,13 @@
  *
  * <p>This is a service provider interface that requires implementations to be registered in a
  * provider-configuration file stored in the {@code META-INF/services} resource directory.
- *
- * @deprecated Use {@link ConfigCustomizer} instead.
  */
-@Deprecated
 public interface ConfigPropertySource extends Ordered {
```

Comment:
is the goal to deprecate this later in favor of AutoConfigurationCustomizer#addPropertiesSupplier?

Reply:
We definitely should do that at some point, probably after we remove all `Config` usages (since the old `Config` won't get the properties added by SDK SPIs)